### PR TITLE
add repr parameter to hide a field from repr()

### PR DESCRIPTION
**Pull request by fnep**
## Change Summary

Add a `repr` parameter to `Field` (defaults to `True`), to hide the field from the object representation with `repr()`.  

## Related issue number

This is related to #1831.

## Checklist

* [x] Unit tests for the changes exist
* [x] Tests pass on CI and coverage remains at 100%
* [x] Documentation reflects the changes where applicable
* [x] `changes/<pull request or issue id>-<github username>.md` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)

### CODE REVIEW

**Comment by epln** [comment]
No clue why the "site-deploy" action is failing.

**Comment by PrettyWood** [comment]
@fnep you can rebase on `master` :)

**Comment by PrettyWood** [comment]
Better update your `master` first and rebase on top. You have a lot of unrelated commits

**Comment by fnep** [comment]
I see something went wrong this time. Sorry, i fixed it.

File: tests/test_main.py [hunk 1]
```diff
@@ -1596,6 +1596,18 @@ class Config:
...
+def test_repr_field():
+    class Model(BaseModel):
+        a: int = Field()
+        b: int = Field(repr=True)
+        c: int = Field(repr=False)
+
+    m = Model(a=1, b=2, c=3)
+    assert 'a=1' in repr(m)
+    assert 'b=2' in repr(m)
+    assert 'c=3' not in repr(m)
```
Comment (PrettyWood) [inline tests/test_main.py:1608]:
```suggestion
    assert 'a=1' in repr(m)
    assert 'b=2' in repr(m)
    assert 'c=3' not in repr(m)
```
```suggestion
    assert repr(m) == 'Model(a=1, b=2)'
    assert repr(m.__fields__['a'].field_info) == 'FieldInfo(default=Ellipsis, extra={})'
    assert repr(m.__fields__['b'].field_info) == 'FieldInfo(default=Ellipsis, extra={})'
    assert repr(m.__fields__['c'].field_info) == 'FieldInfo(default=Ellipsis, repr=False, extra={})'
```
to also test the `field_defaults_to_hide` part

Comment (fnep) [reply tests/test_main.py]: Good point 👍 .

**Review by PrettyWood — APPROVED** [review]
Thanks 👍

**Comment by samuelcolvin** [comment]
great, thanks so much.

**Pull request merged**